### DISCUSSION TITLE
Automatically handling the iOS location permissions popup

### DISCUSSION THREAD

**Discussion by objectivedynamics42** [appium/appium#16915] [Q&A] [answered]
I need to be able to either prevent the location permissions popup from occurring or have it handled by Appium/WebDriver. It looks like the Appium settings API supports these two settings for this purpose:

`appium:autoAcceptAlerts`
and
`acceptAlertButtonSelector`

Unless I'm mistaken, settings have to be applied via a driver instance, something like this (Java):

`driver.setSetting("appium:autoAcceptAlerts",true);`

The problem that I have is that once the driver instance has been created, the application has already launched and the location permissions popup is already on display before we can set the appropriate settings.

I'm sure I must be missing something so I'd appreciate it if anyone can point out my mistake

In case it makes a difference we're using Appium Server 1.22.0 on macOS.

TIA, Mike

**Comment by jlipps** [comment, marked as answer]
`appium:autoAcceptAlerts` is a capability, not a setting, so you can include it in your launch capabilities.

fair warning, it doesn't always work, because there's no guaranteed way of determining that a permissions popup is present.

given that location permissions popups occur at predetermined times in response to actions taken in your app, it's usually better to just conditionally handle popups where they might occur in your test flow.

**Reply by objectivedynamics42** [reply]
Ah, I was basing my comment on it being a setting on its inclusion in the list of settings [on this page](https://github.com/appium/appium-xcuitest-driver#settings-api).
It's also listed as an iOS capability [on this page](https://appium.io/docs/en/writing-running-appium/caps/).
Just so I understand, is it true to say that this is both a setting and a capability?

**Reply by jlipps** [reply]
it looks like `appium:autoAcceptAlerts` is a capability, and `acceptAlertButtonSelector` is a setting. You can also set any setting as a capability by wrapping it in `appium:settings[...]`, so in this case `appium:settings[acceptAlertButtonSelector]` is a capability you could include to set the setting for the whole session.
